jme3-core/src/main/java/com/jme3/math/ColorRGBA.java

```diff
@@ -825,4 +805,25 @@ public ColorRGBA getAsSrgb() {
         srgb.a = a;
         return srgb;
     }
+
+    /**
+     * Helper method to convert a float (0-1) to a byte (0-255).
+     */
+    private byte toByte(float channel) {
+        return (byte) ((int) (channel * 255) & 0xFF);
+    }
+
+    /**
+     * Helper method to convert an int (shifted byte) to a float (0-1).
+     */
+    private float fromByte(int channelByte) {
+        return ((byte) (channelByte) & 0xFF) / 255f;
```

Comment:
I think that this method would need to take in an extra param for the function to do the casting properly so that it returns the correct origianl value (and I think this would also make it more understandable by keeping all of the shifting operation in one line of code within the new method):

    private float fromByte(int color, int shift) {
        return ((byte) (color >> shift) & 0xFF) / 255f;
    }

And if that works to make the new value match the old, then the screenshot tests should also pass. 

(Although i'm also no expert in bit shifting and usually only stumble into bitshifting code every few years when I run into trouble with an internal jme class that uses it, so I inevitably always have to refigure things out with bit shifting anytime I encounter it)